src/algorithms/lbfgs.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <RcppArmadillo.h>
+#include "../families/family.h"
+#include "wolfe_line_search.h"
+
+using namespace Rcpp;
+using namespace arma;
+
+// // LBFGS Algorithm.
+// // Solves argmin_z { f(z) + rho/2*||z-u||^2 }
+template <typename T>
+mat Family::lbfgs(const T& x, const mat&y, const double rho, const mat& u){
+
+  mat z(u);
+  mat g(u);
+  
+  const int max_iter = 2000;
+  const double tolerance = 1e-5;
+  const int l = 20;
+
+  std::deque <mat> dz;
+  std::deque <mat> dgrad;
+  std::deque <double> eta;
+
+  std::vector <double> alpha(l);
```

Comment:
Don't separate template declaration from type
```suggestion
  std::deque<mat> dz;
  std::deque<mat> dgrad;
  std::deque<double> eta;

  std::vector<double> alpha(l);
```